crates/aptos-rosetta/src/types/identifiers.rs

```diff
@@ -222,15 +268,27 @@ impl SubAccountIdentifier {
     }
 
     pub fn is_active_stake(&self) -> bool {
-        self.address.as_str() == ACTIVE_STAKE
+        self.address.as_str() == ACTIVE_STAKE && self.metadata.is_none()
     }
 
     pub fn is_pending_inactive_stake(&self) -> bool {
-        self.address.as_str() == PENDING_INACTIVE_STAKE
+        self.address.as_str() == PENDING_INACTIVE_STAKE && self.metadata.is_none()
     }
 
     pub fn is_inactive_stake(&self) -> bool {
-        self.address.as_str() == INACTIVE_STAKE
+        self.address.as_str() == INACTIVE_STAKE && self.metadata.is_none()
```

Comment:
I understand this is not from this PR but why we have to do this hack? by any means, Neither `STAKE`, `PENDING_INACTIVE_STAKE ` or `INACTIVE_STAKE` is an address. If we just need a enumeration, I believe `metadata` field is for this purpose, right?

Reply:
I feel we probably don't want to deal with extra fields that's why it was initially implemented like this
given the current code is already in use internally i'd prefer just leave it as it is and use the new field get the target data

Reply:
I won't block this PR for this. But I highly suggest to fix it ASAP because it doesn't make sense at all and the schema will be incompatible after fixing.